src/compiler/types.ts

```diff
@@ -3907,6 +3918,8 @@ namespace ts {
         resolvedModule: ResolvedModuleFull | undefined;
         /* @internal */
         failedLookupLocations: string[];
+        /*@internal*/
+        isInvalidated?: boolean;

```

Comment:
Don't think this type should be mutable -- could you use a `{ readonly resolved: ResolvedModuleWithFailedLookupLocations, isInvalidated: boolean }` type for that purpose instead?